### DISCUSSION TITLE
Any plan for global hooks like mocha ?

### DISCUSSION THREAD

**Discussion by ibnumusyaffa** [adonisjs/core#4166] [Japa]
is there any plan for adding global hooks like mocha has? lets say i want to run database migration once before all test run

**Comment by RomainLanz** [comment]
Hey @ibnumusyaffa! 👋🏻 

This is already available and documented.

```ts
export const runnerHooks: Required<Pick<Config, 'setup' | 'teardown'>> = {
  setup: [
    () => TestUtils.ace().loadCommands(),
    () => TestUtils.db().migrate()
  ],
  teardown: [],
}
```

- 📚 [Japa: Lifecycle Hooks](https://japa.dev/docs/lifecycle-hooks)
- 📚 [Adonis: Database Migration during Test](https://docs.adonisjs.com/guides/testing/introduction#migrating-database)
